microsoft/testsuites/core/timesync.py

```diff
@@ -351,3 +352,33 @@ def timesync_chrony(self, node: Node) -> None:
         chrony.check_sources_and_stats()
         # 5. Check chrony tracking.
         chrony.check_tracking()
+
+    @TestCaseMetadata(
+        description="""
+        This test is to check gettime calls are not being redirected as system calls,
+        leading to performance bottleneck, Linux systems have a mechanism called vdso
+        which helps in gettimeoftheday and clock_gettime to be executed in userspace (no
+        syscall).
+
+        Steps:
+            1. Run strace date 2>&1
+            2. Check output of 1, should not show the time related syscalls,
+             e.g. clock_gettime, gettimeofday
+        """,
+        priority=2,
+    )
+    def verify_vdso(self, node: Node) -> None:
+        time_pattern = re.compile("(time|gettimeofday|clock_gettime)", re.M)
```

Comment:
[nit] can be a global variable in the class

Reply:
now change to run vdso benchmark, this code was removed